File: python/429.py

# Definition for a Node.
from typing import List


class Node:
  def __init__(self, val=None, children=None):
    self.val = val
    self.children = children


class Solution:
  def levelOrder(self, root: 'Node') -> List[List[int]]:
    if root:
      q = [root]
      ans = []
      while q:
        n = len(q)
        ans.append([node.val for node in q])
        for i in range(n):
          for child in q[i].children:
            q.append(child)
        q = q[n:]
      return ans
    return []

File: go/429.go
package main

type Node struct {
	Val      int
	Children []*Node
}

func levelOrder(root *Node) (res [][]int) {
	if root == nil {
		return
	}
	st := []*Node{root}
	res_perlevel := []int{}
	for len(st) > 0 {
		res_perlevel = []int{}
		n := len(st)
		for i := 0; i < n; i++ {
			res_perlevel = append(res_perlevel, st[i].Val)
			for _, node := range st[i].Children {
				st = append(st, node)
			}
		}
		res = append(res, res_perlevel)
		st = st[n:]
	}
	return
}
